php/proxy.php

```diff
@@ -3,150 +3,192 @@
+
     private $settings;
     private $targetPath;
     private $username;
     private $password;
-    private $domainURL;	
-	
+    private $domainURL;
+    private $host;
+    private $url;
+
     private function disable_gzip() {
         @ini_set('zlib.output_compression', 'Off');
-        @ini_set('output_handler', '');	
+        @ini_set('output_handler', '');
     }
-    
-    
+
     private function getSettings() {
         $acrolinxSettings = array();
-
         $acrolinxSettings['isSSO'] = 1; //TODO: Set if SSO should be used, otherwise it will be just a reverse proxy.
         $acrolinxSettings['url'] = ""; //TODO: Set Acrolinx server url here.
         $acrolinxSettings['genericPassword'] = "secret"; //TODO: Set secret token here.
 
         if($acrolinxSettings['isSSO'] == 1) {
-
             $acrolinxSettings['password'] = $acrolinxSettings['genericPassword'];
             $acrolinxSettings['username'] = get_current_user(); //TODO: Implement get_current_user, which retrieves the username from the current session.
         }
         return $acrolinxSettings;
     }
-    
+
     private function getPosition($requestURI, $part) {
         return strpos($requestURI, $part);
     }
-    
+
     private function getActualURI($requestURI, $pos, $part) {
         return substr_replace($requestURI, '', 0, ($pos+strlen($part)));
     }
-    
+
+    private function emulate_getallheaders() {
+        foreach ($_SERVER as $name => $value) {
+            if (substr($name, 0, 5) == 'HTTP_') {
+                $name = str_replace(' ', '-', ucwords(strtolower(str_replace('_', ' ', substr($name, 5)))));
+                $headers[$name] = $value;
+            } else if ($name == "CONTENT_TYPE") {
+                $headers["Content-Type"] = $value;
+            } else if ($name == "CONTENT_LENGTH") {
+                $headers["Content-Length"] = $value;
+            }
+        }
+        return $headers;
+    }
+
     private function getHeadersOfRequest() {
         $headers = array();
-        $isAuthToken = false; 
-        foreach (getallheaders() as $name => $value) {
-          $headerString = $name.':'.$value;
-
-          //Setting target Server as Host
-          if(strtolower($name) == 'host') {
-              $headerString = $name.':'.parse_url(self::getSettings()['url'])['host'];
-          }
-
-          if($name == 'authToken') {
-            $isAuthToken = true;
-          }
-          array_push($headers,"$headerString");
+        $isAuthToken = false;
+
+        foreach (self::emulate_getallheaders() as $name => $value) {
+
+            $headerString = $name.':'.$value;
+
+            //Setting target Server as Host
+            if(strtolower($name) == 'host') {
+                $headerString = $name.':'.$this->host;
+            }
+            if($name == 'authToken') {
+                $isAuthToken = true;
+            }
+            array_push($headers,"$headerString");
         }
-        
+
         if($isAuthToken == false){
```

Comment:
I'm wondering if it is a good idea to only add this if no authToken is present? What if the token is invalid? Will it become an endless loop? Sometimes an "authorization" header is used for the token as well (v6 checking)... but sometimes it's not Acrolinx related (basic auth). Also the header will change in the new Acrolinx API.

Reply:
Sure we should fix this. I'll remove this check and let username and password be added for every request.